### PR TITLE
feat: add table description

### PR DESCRIPTION
**Pull request by tiboun**
## Summary
Add table description when creating or getting table

**PR Type: Feature**

**Status: Ready to review**

**Breaking change? No**

### Solution
Refactor table creation and add created a new case class TableInfo to have all information in one structure.

### How has this been tested?
This has been tested with a watch process.

### Other changes
Remove deprecation of ForkJoinPool and clean some codes

## Contributor checklist:
Go over all the following points, and put an `x` in all the boxes that apply.
If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [x] My code follows the code style of this project.
- [x] I have updated the Release notes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/scala/ai/starlake/extract/BigQueryTablesLog.scala [hunk 1]
```diff
@@ -163,11 +169,12 @@ object BigQueryTableLog {
       authInfo.toMap,
       dfTable,
       "table_info",
+      Some("Information related to tables"),
       config.writeMode.getOrElse(WriteMode.OVERWRITE)
     )
   }
 
-  def run(args: Array[String]) = {
+  def run(args: Array[String]): Any = {
```
Comment (hayssams) [inline src/main/scala/ai/starlake/extract/BigQueryTablesLog.scala]: return Unit Please

File: src/main/scala/ai/starlake/extract/BigQueryTablesLog.scala [hunk 2]
```diff
@@ -136,7 +141,7 @@ object BigQueryTableLog {
       info.getRequirePartitionFilter(),
       logTime
     )
-  def sink(config: BigQueryTablesConfig)(implicit settings: Settings) = {
+  def sink(config: BigQueryTablesConfig)(implicit settings: Settings): Any = {
```
Comment (hayssams) [inline src/main/scala/ai/starlake/extract/BigQueryTablesLog.scala]: return Unit please

**Review by hayssams — APPROVED** [review]
LGTM

**Comment by mhdkassir** [comment]
I just started to look into this PR. We are working on an equivalent PR where we use the `Schema.Comment` field to set the Description of the corresponding table instead of creating a new field for that.

**Comment by tiboun** [comment]
> I just started to look into this PR. We are working on an equivalent PR where we use the `Schema.Comment` field to set the Description of the corresponding table instead of creating a new field for that.

Yes, it uses this field. Try it out, it will update straight away your table with the description you have on Schema.Comment. No additional fields have been added in the yml file.

**Comment by hayssams** [comment]
> 
@mhdkassir 
Are you sure you are working on this.
This one is about the domain table. I thought you were working on the job.

**Comment by mhdkassir** [comment]
@hayssams yes we implemented this part on our fork of starlake , description of table and description of columns issued from a job.

**Comment by mhdkassir** [comment]
@tiboun I think that tables created by BigQueryNative Jobs are not covered by this feature, do you confirm ?

**Comment by mhdkassir** [comment]
> @tiboun I think that tables created by BigQueryNative Jobs are not covered by this feature, do you confirm ?

I implemented this part on a ongoing branch, will wait for your confirmation to push it

**Comment by tiboun** [comment]
Yes Mohamad, I confirm what you say because there is no trace of table
creation during job task as far as I know.



Le mar. 21 mars 2023, 15:27, Mohamad Kassir ***@***.***> a
écrit :

**Comment by tiboun** [comment]
> @tiboun I think that tables created by BigQueryNative Jobs are not covered by this feature, do you confirm ?

Yes @mhdkassir , I confirm what you said because I don't see any trace of table creation inside AutoTask. I think you use that in the context of AutoTask, right ?

**Comment by mhdkassir** [comment]
> > @tiboun I think that tables created by BigQueryNative Jobs are not covered by this feature, do you confirm ?
> 
> Yes @mhdkassir , I confirm what you said because I don't see any trace of table creation inside AutoTask. I think you use that in the context of AutoTask, right ?

yes it's an  AutoTask with engine set to BQ. I added the fix in #453

**Pull request merged**